src/UI/Actions.hs

```diff
@@ -1145,35 +1136,49 @@ encapsulateMail =
         case mail of
           Nothing -> showWarning "No mail selected for forwarding"
           Just m -> do
+            charsets <- use (asConfig . confCharsets)
+            let
+              origSubj = views (headerSubject charsets) fold m
+              origFrom = views (headerFrom charsets) AddressText.renderAddresses m
+              newSubj = "[" <> origFrom <> ": " <> origSubj <> "]"
             modifying (asCompose . cAttachments)
               (L.listInsert 1 (encapsulate m) . L.listInsert 0 (createTextPlainMessage mempty))
             modifying (asCompose . cSubject . editEditorL . E.editContentsL)
-              (insertMany (getForwardedSubject m) . clearZipper)
+              (insertMany newSubj . clearZipper)
     }
 
--- | Update the 'AppState' with a quoted form of the first preferred
--- entity in order to reply to the e-mail.
 -- | Update the 'AppState' with a quoted version of the currently
 -- selected mail in order to reply to it.
 --
-replyMail :: Action 'ViewMail 'ScrollingMailView ()
-replyMail = Action
-  { _aDescription = ["reply to an e-mail"]
-  , _aAction = do
+senderReply, groupReply :: Action 'ViewMail 'ScrollingMailView ()
+senderReply = Action ["reply"] (replyWithMode ReplyToSender)
+groupReply = Action ["group-reply"] (replyWithMode ReplyToGroup)
+
+replyWithMode :: ReplyMode -> StateT AppState (T.EventM Name) ()
+replyWithMode mode = do
       mail <- use (asMailView . mvMail)
+      charsets <- use (asConfig . confCharsets)
       case mail of
         Nothing -> do
           modifying (asViews . vsFocusedView) (Brick.focusSetCurrent Threads)
           showWarning "No mail selected for replying"
         Just m -> do
           mailboxes <- use (asConfig . confComposeView . cvIdentities)
+          let
+            idents = case mailboxes of
+              [] -> pure $ Mailbox Nothing (AddrSpec "CHANGE.ME" (DomainDotAtom $ "YOUR" :| ["DOMAIN"]))
```

Comment:
See in this case I'm still not sure what is better: pre-set text or simply nothing ... Both mean that the user is should be aware of what he's doing. Not even sure if we can send a mail with an empty `To` mailbox field now thinking of it...

Reply:
This is for the `From` header.  RFC 5322 requires it, hence I designed `reply` to enforce that also.  And honestly, I don't really see any case where the user wouldn't set their address(es) in the config.

Reply:
I suppose if address is not set in config, when composing or replying we can prompt for it.  Alternatively we could make purebred nag the user if the config isn't set, e.g. show a warning when starting up.  In any case, I'd prefer to deal with that separately from this change.